### ISSUE TITLE
does not gracefully handle optionalDependency "weak"

### ISSUE THREAD

**Issue by jonstorer** [substack/dnode#169]
`weak` is optional. However the code does not gracefully handle the package's optionality.

https://github.com/substack/dnode/blob/master/index.js#L28

**Comment by donaldpipowitch** [comment]
@apsavin Could you look into this?

**Comment by apsavin** [comment]
You can use https://github.com/substack/dnode/blob/master/index.js#L26 option `{weak: false}`, isn't it?

**Comment by donaldpipowitch** [comment]
The error occurs in other modules like https://github.com/sgentle/phantomjs-node/issues/312. I don't know if I can override it there. But it should automatically force `{ weak: false }` if the installation failed and it is really optional.

**Comment by apsavin** [comment]
You can, https://github.com/sgentle/phantomjs-node#use-it-in-windows

**Comment by donaldpipowitch** [comment]
Thanks, I'll try that.

**Comment by donaldpipowitch** [comment]
Seems to work, thank you! But I'd still leave this bug open. Even `{ weak: true }` should not be used, if `weak` fails to install. (It should probably warn about the missing dependency, but not fail, if it is optional.)

**Comment by apsavin** [comment]
Yes, I agree, we can do an improvement here. We can catch the error and provide info how to handle this at least.

**Comment by donaldpipowitch** [comment]
Sounds good!

**Comment by AugustNagro** [comment]
Updating dnode resolved this for me. https://github.com/sgentle/phantomjs-node/issues/307#issuecomment-143140144
